lib/src/common/cache/cache.dart

```diff
@@ -294,6 +301,65 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
+  /// Provides access to a value stored in the cache without informing the
+  /// [CachingStrategy] of a get.
+  ///
+  /// [weakGet] can be thought of sort of like a weak reference when working with
+  /// a garbage collector. [weakGet] won't prevent removal of a [TValue]
+  /// [TIdentifier] pair from the [Cache], [get] or [getAsync] will. In other
+  /// words, it's a get that behaves like `Map`'s does rather than the way
+  /// [Cache]'s does. This can be convenient, for example, when mutating an
+  /// object that is known to be in the cache.
+  ///
+  /// This ideally shouldn't be used to create a reference with a lifetime that
+  /// will exceed the time an item exists in the cache. For example this would be
+  /// a bad consumption pattern:
+  ///
+  /// (when a reference created with [weakGet] would be alive and when an item
```

Comment:
Maybe something like "The delineation to the right of the code indicates when a reference created with [weakGet] would be alive and when an item will be in the cache" would be more clear